### PR TITLE
fix bugs on mixer validator server code

### PR DESCRIPTION
**Pull request by jmuk**
- 'target-namespace' emptiness handling is wrong, the default should
  be 'NamespaceAll'.
- actually the admit.go didn't care about Delete code path -- fixing
  to support it.

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: mixer/pkg/config/crd/admit.go [hunk 1]
```diff
@@ -308,6 +309,7 @@ func (ac *AdmissionController) ServeHTTP(w http.ResponseWriter, r *http.Request)
 			glog.V(4).Infof("Failed to read request body: %v", err)
 		}
 	}
+	glog.V(4).Infof("request body: %s", body)
```
Comment (mandarjog) [inline mixer/pkg/config/crd/admit.go]: why requestBody at V(4) and message on line 302 at V(7) ?

File: mixer/pkg/config/crd/admit.go [hunk 2]
```diff
@@ -360,28 +362,37 @@ func (ac *AdmissionController) admit(review *v1alpha1.AdmissionReview) *v1alpha1
...
-		return makeErrorStatus("cannot decode configuration: %v", err)
-	}
-
-	if !watched(ac.options.ValidateNamespaces, obj.GetNamespace()) {
-		return &v1alpha1.AdmissionReviewStatus{Allowed: true}
-	}
-
-	ev := toEvent(t, &obj)
-	if err := ac.options.Validator.Validate(&ev); err != nil {
+	if err := ac.options.Validator.Validate(ev); err != nil {
```
Comment (mandarjog) [inline mixer/pkg/config/crd/admit.go]: is this where `jsonpb.Parse` is called with the specific proto?

File: mixer/pkg/config/crd/admit.go (hunk 1 above)
Comment (jmuk) [reply mixer/pkg/config/crd/admit.go]: The request body contains the whole data for the validation (i.e. kind, name, namespace, spec, etc.) while I found that the request object itself (in line 302) doesn't show up very helpful information, like request path or http method, and no content information on request body.
I'm fine to remove the line 302.

File: mixer/pkg/config/crd/admit.go (hunk 2 above)
Comment (jmuk) [reply mixer/pkg/config/crd/admit.go]: yes, the implementation of `ac.options.Validator` will be https://github.com/istio/istio/blob/master/mixer/pkg/config/store/validator.go#L55, which conducts the `jsonpb.Parse`. Also, an upcoming PR will do further validation on converted proto objects.

**Pull request merged**